VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -247,10 +247,16 @@ def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, k
         self.logger.log("device items which will be used to find vhds to stamp: {0}".format(all_dev_items_real_paths))
 
         root_device_path = os.path.join(CommonVariables.azure_symlinks_dir, "root")
+        root_device_path_scsi = os.path.join(CommonVariables.azure_symlinks_dir, "scsi0/lun0")
+        is_os_nvme, root_device_path_nvme = disk_util.is_os_disk_nvme()
         if os.path.exists(root_device_path):
             root_device_path = os.path.realpath(root_device_path)
+        elif os.path.exists(root_device_path_scsi):
+            root_device_path = os.path.realpath(root_device_path_scsi)
+        elif is_os_nvme and os.path.exists(root_device_path_nvme):
+            root_device_path = os.path.realpath(root_device_path_nvme)
         else:
-            root_device_path = os.path.realpath(os.path.join(CommonVariables.azure_symlinks_dir, "scsi0/lun0"))
+            self.logger.log("Cannot locate root device")
```

Comment:
If the root device is not found, and we are no longer defaulting to "root_device_path = os.path.realpath(os.path.join(CommonVariables.azure_symlinks_dir, "scsi0/lun0"))", how does the error come across on line 260 if root_device_path is None? 

I can't think of how to test this right now, but it seems there might be a TypeError when root device is not found. Wondering if we want to throw a more specific error ?

Reply:
The default to scsi0/lun0 was breaking in NVMe VM's as it did not exist. An error was thrown when trying to stamp the disks. I believe the same error will be thrown if we are not able to locate the root device.

But, I get the point. I will look into a better error handling probably in the next PR.